api/envoy/api/v2/core/health_check.proto

```diff
@@ -111,6 +116,16 @@ message HealthCheck {
     // <envoy_api_field_core.HealthCheck.HttpHealthCheck.use_http2>` in light of
     // HTTP3.
     type.CodecClientType codec_client_type = 10 [(validate.rules).enum = {defined_only: true}];
+
+    // An optional service name parameter which is used to validate the identity of
+    // the health checked cluster using a :ref:`StringMatcher
+    // <envoy_api_msg_type.matcher.StringMatcher>`. See the :ref:`architecture overview
+    // <arch_overview_health_checking_identity>` for more information.
+    type.matcher.StringMatcher service_name_matcher = 11;
+
+    // An optional parameter to control cluster identity validation using :ref:`StringMatcher
+    // <envoy_api_msg_type.matcher.StringMatcher>`. This is set to true by default.
+    RuntimeFeatureFlag service_name_matcher_enabled = 12;
```

Comment:
I don't think we need an explicit kill switch for this feature in the config, since we already have `"health_check.verify_cluster"` in the code. I think you can just use the existing kill switch to cover the new string matcher also?